bridge-token-factory/src/lib.rs

```diff
@@ -360,33 +400,57 @@ impl BridgeTokenFactory {
 
         let required_deposit = self.record_proof(&proof);
 
-        assert!(
-            env::attached_deposit()
-                >= required_deposit + self.bridge_token_storage_deposit_required
+        let Recipient { target, message } = parse_recipient(new_owner_id);
+
+        let token = EthAddressHex(token);
+        let amount_to_transfer: u128;
+        let fee_amount = self.calculate_deposit_fee_amount(
+            &token,
+            amount,
+            message.as_ref().and_then(|_| Some(&target)),
         );
+        amount_to_transfer = amount - fee_amount;
```

Comment:
Looks incorrect for me. 

Current fee calculation: 
fee_amount = amount *  fee_percentage =  (amount_to_transfer + fee_amount) *  fee_percentage

More logical: 
fee_amount = amount_to_transfer * fee_percentage
fee_amount + fee_amount*fee_percentage = amount * fee_percentage
fee_amount = amount * fee_percentage/(1 + fee_percentage)